### ISSUE TITLE
[4.2] Auth::onceBasic() thinks the user is unauthenticated when accessed

### ISSUE THREAD

**Issue by cwhite92** [laravel/framework#6328]
Building a RESTful API using HTTP basic authentication, using `Auth::onceBasic()` in my filter like so:

    Route::filter('auth.api', function() {
    	$result = Auth::onceBasic();

    	return $result;
    });

Protecting my controller routes like so:

	public function __construct()
	{
		$this->beforeFilter('auth.api', ['except' => 'store']);
	}

This works fine, with correct user credentials I can access the authenticated user using `Auth::user()` as normal. When specifying incorrect user credentials I get the `Invalid credentials.` response. I want to change the invalid credentials response to be consistent with the rest of my API, so I changed my filter to this:

	Route::filter('auth.api', function() {
		$result = Auth::onceBasic();
	
		if ($result->getStatusCode() === 401) {
			// Unauthorized, return our own response
			return Response::json([
				'message' => 'Bad credentials'
			], Config::get('status.error.unauthorized'));
		}
	
		return $result;
	});

Now my custom response gets returned every time, regardless of if the user's credentials are correct. I'm specifying the user credentials in the URL like this:

    http://user@gmail.com:testpass@project.app/users/2

TLDR: Auth::onceBasic() seems to work in the filter until you try to use it, in my case to get the status code, at which point is suddenly thinks the user is unauthenticated.

EDIT: I should specify that this happens in Laravel version 4.2.7

**Comment by GrahamCampbell** [comment]
> EDIT: I should specify that this happens in Laravel version 4.2.7

Try with 5.0.1.
